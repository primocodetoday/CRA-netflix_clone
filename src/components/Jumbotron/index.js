import React from 'react';
import PropTypes from 'prop-types';
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
} from './styles/jumbotron';

const Jumbotron = ({ children, direction }, ...restProps) => {
  return (
    <Item {...restProps} direction={direction}>
      <Inner>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumboSubtronTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Jumbotron.propTypes = {
  direction: PropTypes.string,
};

Jumbotron.defaultProps = {
  direction: 'row',
};

export default Jumbotron;
