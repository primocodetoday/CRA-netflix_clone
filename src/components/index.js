export { default as Footer } from 'components/Footer';
export { default as Accordion } from 'components/Accordion';
export { default as Jumbotron } from 'components/Jumbotron';
