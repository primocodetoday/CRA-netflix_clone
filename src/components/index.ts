export { Accordion } from 'components/Accordion/Accordion';
export { Banner } from 'components/Banner/Banner';
export { Card } from 'components/Card/Card';
export { CTAForm } from 'components/CTAForm/CTAForm';
export { Form } from 'components/Form/Form';
export { Footer } from 'components/Footer/Footer';
export { Header } from 'components/Header/Header';
export { Jumbotron } from 'components/Jumbotron/Jumbotron';
export { Loading } from 'components/Loading/Loading';
export { Player } from 'components/Player/Player';
export { Profiles } from 'components/Profiles/Profiles';
