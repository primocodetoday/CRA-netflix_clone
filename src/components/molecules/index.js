export { default as Footer } from 'components/molecules/Footer/Footer';
export { default as Accordion } from 'components/molecules/Accordion/Accordion';
export { default as Jumbotron } from 'components/molecules/Jumbotron/Jumbotron';
export { default as CTAForm } from 'components/molecules/CTAForm/CTAForm';
export { default as Header } from 'components/molecules/Header/Header';
