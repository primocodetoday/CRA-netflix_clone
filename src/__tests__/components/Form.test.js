import React from 'react';
import { render } from '@testing-library/react';
import { Form } from 'components';

jest.mock('react-router-dom');

describe('<Form/> renders properly', () => {
  it('with data', () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign</Form.Title>
        <Form.Base>
          <Form.Input placeholder="First Name" />
          <Form.Submit disabled type="submit">
            Sign Up
          </Form.Submit>
          <Form.Text>
            Already a user? <Form.Link to="/signin">Sign in now.</Form.Link>
          </Form.Text>
          <Form.TextSmall>This page is protected by Google</Form.TextSmall>
        </Form.Base>
      </Form>,
    );

    expect(getByText('This page is protected by Google')).toBeTruthy();
    expect(getByText('Sign')).toBeTruthy();
    expect(getByText('Sign Up').disabled).toBeTruthy();
    expect(getByPlaceholderText('First Name')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with an error', () => {
    const error = 'You name not match pattern';
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>{error}</Form.Error>
        <Form.Submit type="submit">Sign In</Form.Submit>
      </Form>,
    );

    expect(getByText(error)).toBeTruthy();
    expect(queryByText('Sign In').disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
