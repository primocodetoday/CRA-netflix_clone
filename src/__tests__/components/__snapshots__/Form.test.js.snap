// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> renders properly with an error 1`] = `
<div
  class="StyledForm__Container-sc-1t60tbe-0 jHRjtY"
>
  <div
    class="StyledForm__Error-sc-1t60tbe-1 lhVaYd"
  >
    You name not match pattern
  </div>
  <button
    class="StyledForm__Submit-sc-1t60tbe-8 ITnHm"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Form/> renders properly with data 1`] = `
<div
  class="StyledForm__Container-lxzb7x-0 iPaOuG"
>
  <h1
    class="StyledForm__Title-lxzb7x-3 kcoXtS"
  >
    Sign
  </h1>
  <form
    class="StyledForm__Base-lxzb7x-2 hUTmBx"
  >
    <input
      class="StyledForm__Input-lxzb7x-7 hNDNrQ"
      placeholder="First Name"
    />
    <button
      class="StyledForm__Submit-lxzb7x-8 iNyBrA"
      disabled=""
      type="submit"
    >
      Sign Up
    </button>
    <p
      class="StyledForm__Text-lxzb7x-4 iYsoVD"
    >
      Already a user? 
    </p>
    <p
      class="StyledForm__TextSmall-lxzb7x-5 jnhySP"
    >
      This page is protected by Google
    </p>
  </form>
</div>
`;
